abstractions/dotnet/src/RequestInfo.cs

```diff
@@ -20,6 +20,25 @@ public class RequestInfo
         /// </summary>
         public Uri URI { get; set; }
         /// <summary>
+        /// Sets the URI of the request.
+        /// </summary>
+        /// <param name="currentPath">the current path (scheme, host, port, path, query parameters) of the request.</param>
+        /// <param name="pathSegment">the segment to append to the current path.</param>
+        /// <param name="isRawUrl">whether the path segment is a raw url. When true, the segment is not happened and the current path is parsed for query parameters.</param>
+        public void SetURI(string currentPath, string pathSegment, bool isRawUrl) {
+            if (isRawUrl) {
```

Comment:
nit: Inconsistent formatting with the rest of the file with the opening braces showing up in the same line. 😅

Reply:
```suggestion
            if (isRawUrl)
            {
```